tests/fixture/testnet/local/node.go

```diff
@@ -0,0 +1,295 @@
+		return err
+	}
+	n.NodeConfig = config
+	return nil
+}
+
+func (n *LocalNode) WriteConfig() error {
+	if err := os.MkdirAll(n.GetDataDir(), perms.ReadWriteExecute); err != nil {
+		return fmt.Errorf("failed to create node dir: %w", err)
+	}
+
+	bytes, err := testnet.DefaultJSONMarshal(n.Flags)
+	if err != nil {
+		return fmt.Errorf("failed to marshal local node config: %w", err)
+	}
+
+	if err := os.WriteFile(n.GetConfigPath(), bytes, perms.ReadWrite); err != nil {
+		return fmt.Errorf("failed to write local node config: %w", err)
+	}
+	return nil
+}
+
+func (n *LocalNode) GetProcessContextPath() string {
+	return filepath.Join(n.GetDataDir(), cfg.ProcessContextFilename)
+}
+
+func (n *LocalNode) ReadProcessContext() error {
+	path := n.GetProcessContextPath()
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// The absence of the process context file indicates the node is not running
+		return nil
+	}
+
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read local node process context: %w", err)
+	}
+	processContext := node.NodeProcessContext{}
+	if err := json.Unmarshal(bytes, &processContext); err != nil {
+		return fmt.Errorf("failed to unmarshal local node process context: %w", err)
+	}
+	n.NodeProcessContext = processContext
+	return nil
+}
+
+func (n *LocalNode) ReadAll() error {
+	if err := n.ReadConfig(); err != nil {
+		return err
+	}
+	if err := n.ReadProcessContext(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *LocalNode) Start(w io.Writer, defaultExecPath string) error {
+	// Ensure a stale process context file is removed so that the
+	// creation of a new file can indicate node start.
+	if err := os.Remove(n.GetProcessContextPath()); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove stale process context file: %w", err)
+	}
+
+	execPath := n.ExecPath
+	if len(execPath) == 0 {
+		execPath = defaultExecPath
+	}
+
+	cmd := exec.Command(execPath, "--config-file", n.GetConfigPath())
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			if err.Error() != "signal: killed" {
+				_, _ = fmt.Fprintf(w, "node %q finished with error: %v\n", n.NodeID, err)
+			}
+		} else {
+			_, _ = fmt.Fprintf(w, "node %q finished: %v\n", n.NodeID, cmd.ProcessState.String())
+		}
+	}()
+
+	if _, err := fmt.Fprintf(w, "Started %s\n", n.NodeID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *LocalNode) Stop() error {
+	pid := n.PID
+	if pid == 0 {
+		// Nothing to do
+		return nil
+	}
+	proc, err := os.FindProcess(pid)
+	if err != nil {
+		return fmt.Errorf("failed to find process to stop: %w", err)
+	}
+	// Sending 0 will not actually send a signal but will perform
+	// error checking. If no error is returned, the process is still
+	// running.
+	if err := proc.Signal(syscall.Signal(0)); err != nil {
+		if strings.Contains(err.Error(), "os: process already finished") {
+			return nil
+		}
```

Comment:
This check makes me sad :( - any better way to check the error? (I see we do some different error checking below?)

Reply:
`os.ErrProcessDone`
```
if errors.Is(err, os.ErrProcessDone) {
  return nil
}
```